bazel/emscripten_toolchain/crosstool.bzl

```diff
@@ -1099,12 +1105,22 @@ def _impl(ctx):
         cc_target_os = cc_target_os,
     )
 
-emscripten_cc_toolchain_config_rule = rule(
+_emscripten_cc_toolchain_config_rule = rule(
     implementation = _impl,
     attrs = {
         "cpu": attr.string(mandatory = True, values = ["asmjs", "wasm"]),
         "em_config": attr.label(mandatory = True, allow_single_file = True),
         "emscripten_binaries": attr.label(mandatory = True),
+        "is_windows": attr.bool(mandatory = True),
     },
     provides = [CcToolchainConfigInfo],
 )
+
+def emscripten_cc_toolchain_config_rule(**kwargs):
```

Comment:
this seems unnecessary to hide in a macro and somewhat obscures its purpose. 

could you just make an attribute `script_extension` which is set in BUILD.bazel and uses a similar select to set it?

Reply:
Sorry if I'm misunderstanding you, but wouldn't having a `script_extension` attribute put the burden of setting it on the user? The extension used (`.bat` vs `.sh`) is more or less an internal detail and less of a useful configuration option in my eyes.

Do you know of a way to set the attributes default similar to what the select is doing? That could be a nice middle ground. AFAIK attributes don't allow defaults to be set with select (https://github.com/bazelbuild/bazel/issues/287).

This strategy for checking for windows is used in similar fashion in [rules_nodejs](https://github.com/bazelbuild/rules_nodejs) and [bazel skylib](https://github.com/bazelbuild/bazel-skylib). e.g. https://github.com/bazelbuild/rules_nodejs/blob/aa09b5796faaf2470b84246a8f00e6e27181c63b/internal/common/copy_to_bin.bzl#L40-L65

Reply:
> wouldn't having ... attribute put the burden of setting on the user?

I don't think so--unless they typically modify the BUILD.bazel that also ships in this repo? 

You're right, attributes can't be set with a select default. I think the "correct" way to handle this would be to make this part of a `toolchain()` that bazel selects for us, but I'm not going to push for that. 

Given that this is how rules_nodejs and bazel skylib do it though, I'm content to keep it consistent, so this is fine as-is.

Reply:
I took a look again and  realized use of `emscripten_cc_toolchain_config_rule` would be unlikely from users. I added the `script_extension` per your suggestion.

Reply:
LGTM thanks